### PR TITLE
Authorized particular user to do particular task

### PR DESCRIPTION
**Pull request by Tarun-Rajj**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -11,8 +11,7 @@
 load_dotenv()
 
 import jwt
-from datetime import datetime
-
+import datetime
```
Comment (AayushPyAI) [inline app.py:14]: why  you decide to use `datetime` instead of `from datetime import datetime` ?

File: app.py [hunk 2]
```diff
@@ -27,7 +26,7 @@
 
 
 # User Schema
-user_schema = {
+user_schema = {
```
Comment (AayushPyAI) [inline app.py:29]: Avoid unwanted spaces

File: app.py [hunk 3]
```diff
@@ -84,6 +84,9 @@ def login():
         user = mongo.db.users.find_one({'username': data['username']})
         if user and check_password_hash(user['password'], data['password']):
             # Generate JWT token
+            # expiration = datetime.datetime.utcnow() + datetime.timedelta(days=1)  # Token expires in 1 day
+            # access_token = jwt.encode({'username': data['username'], 'exp': expiration}, app.config['SECRET_KEY']== os.getenv('SECRET_KEY'))     #original one
```
Comment (AayushPyAI) [inline app.py:88]: Remove unused codes

File: app.py [hunk 4]
```diff
@@ -107,9 +111,14 @@ def add_task():
         data = request.get_json()
         if data.get('name') == '' or data.get('description') == '':
             return jsonify({'error': 'Name and description cannot be blank'}), 400
+        task={
             
-        # Create the task dictionary
-        task = {k: task_schema[k](v) for k, v in data.items() if k in task_schema}
+            'name': data['name'],
+            'description': data['description'],
+            'completed': data.get('completed',False),
```
Comment (AayushPyAI) [inline app.py:118]: I asked to remove it , why it is added again?
'completed': data.get('completed',False),

File: app.py [hunk 5]
```diff
@@ -118,35 +127,53 @@ def add_task():
...
+    try:
+
+        current_user= get_jwt_identity()
+        tasks = list(mongo.db.tasks.find({'user': current_user}))
+        formatted_tasks = [{'_id': str(task['_id']), 'name': task.get('name',None)} for task in tasks]
 
-    return jsonify({'tasks':formatted_tasks})
+        return jsonify({'tasks':formatted_tasks})
+    except Exception as e:
+        return jsonify({'error':str(e)})
```
Comment (AayushPyAI) [inline app.py:146]: No need for try-except here

File: app.py [hunk 6]
```diff
@@ -118,35 +127,53 @@ def add_task():
...
+    try:
+        current_user = get_jwt_identity()
+        task = mongo.db.tasks.find_one({'_id': ObjectId(task_id),'user': current_user})
+        if task:
+            formatted_task = {'_id': str(task['_id']), 'name': task['name'], 'description': task['description']}
+            return jsonify({'task': formatted_task})
+        else:
+            return jsonify({'message': 'Task not found or unauthorized'}), 404
+    except Exception as e:
+        return jsonify({'error':str(e)})
```
Comment (AayushPyAI) [inline app.py:162]: No need for try-except here

File: app.py [hunk 7]
```diff
@@ -158,16 +185,24 @@ def update_task(task_id):
...
+    try:
+        current_user=get_jwt_identity()
+        task = mongo.db.tasks.find_one({'_id':ObjectId(task_id),'user':current_user})
+        if task:
+            mongo.db.tasks.delete_one({'_id': ObjectId(task_id)})
+            return jsonify({'message': 'Task deleted successfully'})
+        else:
+            return jsonify({'message': 'Task not found or unauthorized'}),404
+    except Exception as e:
+            return jsonify({'error':str(e)})
```
Comment (AayushPyAI) [inline app.py:205]: No Need for try-except here also

File: app.py (hunk 1 above)
Comment (Tarun-Rajj) [reply app.py:14]: Sir both are correct but previously this module gives an error so that's why I tried to change it.

**Pull request merged**